### PR TITLE
Logging and error handling audit

### PR DESCRIPTION
**Pull request by dnkolegov**

### CODE REVIEW

**Review by matejpavlovic — APPROVED** [review]
👍

File: pkg/simplewal/simplewal.go [hunk 1]
```diff
@@ -150,7 +151,7 @@ func (w *WAL) loadAll(forEach func(index t.RetentionIndex, p *eventpb.Event)) er
 
 	for i := firstIndex; i <= lastIndex; i++ {
 		data, err := w.log.Read(i)
-		if err != nil {
+		if err != nil && !errors.Is(err, io.EOF) {
```
Comment (matejpavlovic) [inline pkg/simplewal/simplewal.go:154]: I think EOF is still an error here. The log contains metadata about the `lastIndex` and that should be present. If it's not, it's an error.

Also, on line 149, there seems to be a copy-paste error. I believe it should sah "could not read last index".

File: pkg/systems/smr/appmodule.go [hunk 2]
```diff
@@ -30,8 +31,9 @@ type AppModule struct {
 }
 
 // NewAppModule creates a new AppModule.
-func NewAppModule(appLogic AppLogic, transport net.Transport, protocolModule t.ModuleID) *AppModule {
+func NewAppModule(ctx context.Context, appLogic AppLogic, transport net.Transport, protocolModule t.ModuleID) *AppModule {
```
Comment (matejpavlovic) [inline pkg/systems/smr/appmodule.go:34]: I'd prefer not using the context here and keep the TODO context when connecting the transport. That TODO context should eventually disappear and we don't have to pollute the `NewAppModule` interface.
Plus, it tends to be bad practice to store context in state variables.

File: pkg/iss/pbftviewchange.go [hunk 3]
```diff
@@ -588,13 +589,13 @@ func reconstructPSetQSet(
 
 		pSet, err = reconstructPSet(svc.ViewChange.PSet)
 		if err != nil {
-			// TODO: Log warning here
+			logger.Log(logging.LevelWarn, "reconstruct P set err", "err", err)
```
Comment (matejpavlovic) [inline pkg/iss/pbftviewchange.go:592]:
```suggestion
			logger.Log(logging.LevelWarn, "could not reconstruct PSet for PBFT view change", "err", err)
```

File: pkg/iss/pbftviewchange.go [hunk 4]
```diff
@@ -588,13 +589,13 @@ func reconstructPSetQSet(
...
 			continue
 		}
 
 		qSet, err = reconstructQSet(svc.ViewChange.QSet)
 		if err != nil {
-			// TODO: Log warning here
+			logger.Log(logging.LevelWarn, "reconstruct Q set err", "err", err)
```
Comment (matejpavlovic) [inline pkg/iss/pbftviewchange.go:598]:
```suggestion
			logger.Log(logging.LevelWarn, "could not reconstruct QSet for PBFT view change", "err", err)
```

**Review by matejpavlovic — APPROVED** [review]
Nice, thanks! Just minor comments.

File: pkg/simplewal/simplewal.go (hunk 1 above)
Comment (dnkolegov) [reply pkg/simplewal/simplewal.go:154]: fixed

File: pkg/systems/smr/appmodule.go (hunk 2 above)
Comment (dnkolegov) [reply pkg/systems/smr/appmodule.go:34]: TODO context should disappear or the context should disappear at all?

Comment (dnkolegov) [reply pkg/systems/smr/appmodule.go:34]: Can we address this context properly now but not eventually?

Comment (dnkolegov) [reply pkg/systems/smr/appmodule.go]: Fixed. Removed all contexts

**Review by matejpavlovic — APPROVED** [review]
Good job, thanks!

**Pull request merged**